[2025-06-27T05:30:03.141+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-06-27T05:30:03.187+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_processing_pipeline.Structure_Data_With_Spark manual__2025-06-27T05:24:54.789753+00:00 [queued]>
[2025-06-27T05:30:03.195+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_processing_pipeline.Structure_Data_With_Spark manual__2025-06-27T05:24:54.789753+00:00 [queued]>
[2025-06-27T05:30:03.196+0000] {taskinstance.py:2306} INFO - Starting attempt 2 of 2
[2025-06-27T05:30:03.759+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): Structure_Data_With_Spark> on 2025-06-27 05:24:54.789753+00:00
[2025-06-27T05:30:03.763+0000] {standard_task_runner.py:64} INFO - Started process 179 to run task
[2025-06-27T05:30:03.767+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'spark_processing_pipeline', 'Structure_Data_With_Spark', 'manual__2025-06-27T05:24:54.789753+00:00', '--job-id', '38', '--raw', '--subdir', 'DAGS_FOLDER/dag.py', '--cfg-path', '/tmp/tmp5vocw5mg']
[2025-06-27T05:30:03.770+0000] {standard_task_runner.py:91} INFO - Job 38: Subtask Structure_Data_With_Spark
[2025-06-27T05:30:03.914+0000] {task_command.py:426} INFO - Running <TaskInstance: spark_processing_pipeline.Structure_Data_With_Spark manual__2025-06-27T05:24:54.789753+00:00 [running]> on host 83875e1aa1c5
[2025-06-27T05:30:04.100+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spark_processing_pipeline' AIRFLOW_CTX_TASK_ID='Structure_Data_With_Spark' AIRFLOW_CTX_EXECUTION_DATE='2025-06-27T05:24:54.789753+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-06-27T05:24:54.789753+00:00'
[2025-06-27T05:30:04.101+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-06-27T05:30:04.147+0000] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2025-06-27T05:30:04.148+0000] {spark_submit.py:401} INFO - Spark-Submit cmd: /home/***/.local/lib/python3.10/site-packages/pyspark/bin/spark-submit --master yarn --conf spark.master=spark://spark-master:7077 --name arrow-spark --verbose --queue root.default /opt/***/include/scripts/structured_data.py /opt/***/data/raw/raw.json /opt/***/data/foundation
[2025-06-27T05:30:05.158+0000] {spark_submit.py:571} INFO - Using properties file: null
[2025-06-27T05:30:05.234+0000] {spark_submit.py:571} INFO - Exception in thread "main" org.apache.spark.SparkException: When running with master 'yarn' either HADOOP_CONF_DIR or YARN_CONF_DIR must be set in the environment.
[2025-06-27T05:30:05.234+0000] {spark_submit.py:571} INFO - at org.apache.spark.deploy.SparkSubmitArguments.error(SparkSubmitArguments.scala:650)
[2025-06-27T05:30:05.235+0000] {spark_submit.py:571} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateSubmitArguments(SparkSubmitArguments.scala:281)
[2025-06-27T05:30:05.235+0000] {spark_submit.py:571} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateArguments(SparkSubmitArguments.scala:237)
[2025-06-27T05:30:05.235+0000] {spark_submit.py:571} INFO - at org.apache.spark.deploy.SparkSubmitArguments.<init>(SparkSubmitArguments.scala:122)
[2025-06-27T05:30:05.235+0000] {spark_submit.py:571} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2$$anon$3.<init>(SparkSubmit.scala:1103)
[2025-06-27T05:30:05.236+0000] {spark_submit.py:571} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.parseArguments(SparkSubmit.scala:1103)
[2025-06-27T05:30:05.236+0000] {spark_submit.py:571} INFO - at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:86)
[2025-06-27T05:30:05.236+0000] {spark_submit.py:571} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1120)
[2025-06-27T05:30:05.236+0000] {spark_submit.py:571} INFO - at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1129)
[2025-06-27T05:30:05.237+0000] {spark_submit.py:571} INFO - at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
[2025-06-27T05:30:05.251+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-06-27T05:30:05.277+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 502, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: /home/***/.local/lib/python3.10/site-packages/pyspark/bin/spark-submit --master yarn --conf spark.master=spark://spark-master:7077 --name arrow-spark --verbose --queue root.default /opt/***/include/scripts/structured_data.py /opt/***/data/raw/raw.json /opt/***/data/foundation. Error code is: 1.
[2025-06-27T05:30:05.281+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=spark_processing_pipeline, task_id=Structure_Data_With_Spark, run_id=manual__2025-06-27T05:24:54.789753+00:00, execution_date=20250627T052454, start_date=20250627T053003, end_date=20250627T053005
[2025-06-27T05:30:05.548+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 38 for task Structure_Data_With_Spark (Cannot execute: /home/***/.local/lib/python3.10/site-packages/pyspark/bin/spark-submit --master yarn --conf spark.master=spark://spark-master:7077 --name arrow-spark --verbose --queue root.default /opt/***/include/scripts/structured_data.py /opt/***/data/raw/raw.json /opt/***/data/foundation. Error code is: 1.; 179)
[2025-06-27T05:30:05.584+0000] {local_task_job_runner.py:243} INFO - Task exited with return code 1
[2025-06-27T05:30:05.623+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-06-27T05:30:06.071+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
